flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -618,32 +623,51 @@ public SavepointFetchResult fetchSavepointInfo(
             savepointStatusMessageParameters.jobIdPathParameter.resolve(JobID.fromHexString(jobId));
             savepointStatusMessageParameters.triggerIdPathParameter.resolve(
                     TriggerId.fromHexString(triggerId));
-            CompletableFuture<AsynchronousOperationResult<SavepointInfo>> response =
-                    clusterClient.sendRequest(
-                            savepointStatusHeaders,
-                            savepointStatusMessageParameters,
-                            EmptyRequestBody.getInstance());
-
-            if (response.get() == null || response.get().resource() == null) {
-                return SavepointFetchResult.pending();
-            }
-
-            if (response.get().resource().getLocation() == null) {
-                if (response.get().resource().getFailureCause() != null) {
-                    LOG.error(
-                            "Failure occurred while fetching the savepoint result",
-                            response.get().resource().getFailureCause());
-                    return SavepointFetchResult.error(
-                            response.get().resource().getFailureCause().toString());
-                } else {
+            while (true) {
+                CompletableFuture<AsynchronousOperationResult<SavepointInfo>> response =
+                        clusterClient.sendRequest(
+                                savepointStatusHeaders,
+                                savepointStatusMessageParameters,
+                                EmptyRequestBody.getInstance());
+
+                if (response.get() == null || response.get().resource() == null) {
                     return SavepointFetchResult.pending();
                 }
+
+                if (response.get().resource().getLocation() == null) {
+                    // If the failure cause contains 'Not all required tasks are currently
+                    // running.', then continue until savepoint is successfully fetched.
+                    if (response.get().resource().getFailureCause() != null) {
```

Comment:
Once we get back the error `Aborting checkpoint. Failure reason: Not all required tasks are currently running.`, the checkpoint will never succeed.

Reply:
Actually this error happens when the tasks are coming up. Checking the job status only, even when it is RUNNING state won't be sufficient.

Reply:
@morhidi, the main idea is that if the job status is running but the error `Aborting checkpoint. Failure reason: Not all required tasks are currently running.`, we could retry to trigger checkpoint until it's successfully triggered. IMO,  this error could happen when certain tasks are initialized.